scripts/launch-ec2

```diff
@@ -464,7 +464,7 @@ def main():
     region_zone = '{0}{1}'.format(args.region, args.zone)
     try:
         vpc_id, subnet, routetable = None, None, None
-        if not args.no_vpc:
+        if not args.use_default_networking:
```

Comment:
If it's default, then we should use it by default (I think).  Can we say --use-classic-networking?  t1, m1, and other "classic" instance types support classic networking;  t2 and all "current" generation instances don't support classic and default to vpc.  Is it possible launch a t2 without vpc?  and is that classic (I'm told classic is not available on t2 and current stuff).

Reply:
The behaviour of `launch-ec2` without this flag is to create a VPC specifically for cloud-init tests and launch instances in there; I think that's sensible default behaviour for our tool, as it reduces interference with other usage of the AWS account.

The "default" being referred to in the option name is the default networking for your AWS account, which will be either classic networking (if you have a very old AWS account, or have asked AWS to monkey with your account so you have classic) or an Amazon-created "default" VPC.

(`--use-classic-networking` would be inaccurate (or require us to perform checks as to what default networking your account uses), because most AWS accounts can't use classic networking at all.)

Reply:
I see what you're saying; however I think it relies on the user account and changes behavior accordingly; I think that can be confusing and would prefer to be declarative with type of networking needed for the instance (vpc can be default and --use-classic-networking as the alternative) and confirm if it can be done (or not).  We can then fail before launching and running a test in a network mode that isn't what users intended.

Reply:
Done.